lib/matplotlib/widgets.py

```diff
@@ -329,7 +329,9 @@ class Slider(SliderBase):
     def __init__(self, ax, label, valmin, valmax, valinit=0.5, valfmt=None,
                  closedmin=True, closedmax=True, slidermin=None,
                  slidermax=None, dragging=True, valstep=None,
-                 orientation='horizontal', *, initcolor='r', **kwargs):
+                 orientation='horizontal', *, initcolor='r',
+                 track_color='lightgrey', handle_facecolor='white',
```

Comment:
I'm contemplating whether we should collect all handle attributes into one parameter, in the form
`handle={'color': 'lightgrey', facecolor='white', edgecolor='.75', size=10}`. That's a bit more indirect and less discoverable (we need to write down the supported keys in the docstring).

But it would be easier to expand on this later. E.g. we could use `handle='none'`, or `handle={'type': 'rect'}`.

The decision depends a bit on whether we assume that further options will be added.

Reply:
In this case the default argument would become `None`?

I'm a fan of adding flexibility here, especially because the handle is a private attribute and predicting the future is well known to be a fraught activity. As for it being indirect, that is indeed unfortunate, but I suspect that the proposed slider styling example will go a long way towards fixing that.

Are the other examples of an argument like this that I can use for inspiration for writing the docstring?  (Maybe there should be a short blurb somewhere in the contributing docs explaining how to approach this and other non-trivial docstrings?)

Reply:
The examples we have are unfortunately not very well documented:

- the parameter `fontdict` e.g. in [set_title](https://matplotlib.org/devdocs/api/_as_gen/matplotlib.axes.Axes.set_title.html)
- the `bbox` property of [Text](https://matplotlib.org/devdocs/api/text_api.html#matplotlib.text.Text)
- the `arrowprops` in [annotate](https://matplotlib.org/devdocs/api/_as_gen/matplotlib.axes.Axes.annotate.html)

I'd do it like this:

~~~
        handle : dict
            Properties of the slider handle. Supported values are

            ========= ===== ======= ========================================
            Key       Value Default Description
            ========= ===== ======= ========================================
            facecolor color 'white' The facecolor of the slider handle.
            edgecolor color '.75'   The edgecolor of the slider handle.
            size      int   10      The size of the slider handle in points.
            ========= ===== ======= ========================================

~~~
I hope it's valid ReST - didn't check - but you get the point.

Reply:
Got it thanks!

> Supported values are

Should we be limiting this? I was imagining that this would be passed as `ax.plot(..., **handle)` that way if anyone feels very passionate about restyling the handle they can do whatever they want.

Reply:
You need to `marker_props = {f'marker{key}': val for key, val in handle}` to be able to `ax.plot(..., **marker_props)`, which effectively is a strong limit. If you want a better error message, you'd need explicit checks. In that case, you should write a function `check_dict_keys()` similar to the existing `check_*` methods.

Reply:
I'm not sure I fully understood your last comment, but I decided to go with your initial suggestion. I think it makes the docstring far clearer to the user.

Reply:
What I was trying to say: You are converting the keys in handle to 'marker{key}' and pass that to `plot()`. Thus, only keys for which a 'marker{key}' is a valid parameter to `plot()` will be accepted. This already limits the possible handle parameters to more or less reasonable values.

If you passe `handle={'foo': 1}`, you get a message 
> 'Line2D' object has no property 'markerfoo'

which is fair but not overly precise. In case you want a better error message such as "'foo' is not a valid handle parameter", you'd need to validate the handle explicitly.

Reply:
Thank you for explaining! I've tried include this in the docstrings.